wire-library/wire-tests/src/commonTest/proto/java/custom_options.proto

```diff
@@ -115,8 +119,8 @@ extend FooBar {
 
 message MessageWithOptions {
   option (my_message_option_one).foo = 1234;
-  option (my_message_option_one.bar) = "5678";
-  option (my_message_option_one.baz.value) = BAZ;
+  option (my_message_option_one).bar = "5678";
```

Comment:
When interpreting the options, everything inside parenthesis is the name of an extension. Subsequent field accesses must be outside the parentheses. Subsequent field access can actually even refer to an extension, if it also has parentheses:
```protobuf
option (cusotm.option).msg_field.(extension.for.msg_field) = "abc";
```
The actual error from `protoc`:
```
custom_options.proto:118:10: Option "(my_message_option_one.bar)" unknown. Ensure that your proto definition file imports the proto which defines the option.
```

Reply:
We've been reverse engineering but hastily. Love that you know how it's supposed to work.

Reply:
Yep, I understand. I think a lot of the issues I see in Wire were also present in my earlier attempts in [protoparse](https://pkg.go.dev/github.com/jhump/protoreflect/desc/protoparse).

At this point, I've made dozens of changes over the past two years (some non-trivial) to make it nearly perfectly match the output of `protoc`. A lot of the fine-tuning had to be based on actually reading all of the `protoc` code -- especially for some of the weirder stuff it does, like resolving relative references (in type and extension names), interpreting and serializing custom options, and the areas where the actual parser varies from the published grammars on the developer docs web sites. I've only got a handful of remaining gaps (7 to be precise, and 2 of those are actually due to bugs in `protoc` that I'm hoping will be accepted as fixes over there.)

I'm actually in the process of documenting the language, as a real spec (instead of the protoc C++ source code being the de-facto spec). Stay tuned!